### ISSUE TITLE
Create only one certificate per domain for multiple routes with same domain

### ISSUE THREAD

**Issue by mguillem** [ibotty/openshift-letsencrypt#14]
According to the log (and my understanding of the code), it seems that a new certificate is requested for each route, even if the host name is the same. This is a problem due to the rate limit of letsencrypt.

In our case, multiple routes are used to map different services to sub paths.

**Comment by ibotty** [comment]
Yes, that's right. Any idea how an api should look like? I prefer being explicit which domains to combine.

**Comment by ibotty** [comment]
Oh, you mean routes with paths. Yeah, that's a bug, I introduced when going from secrets to just storing the certificates in the route.

**Comment by mguillem** [comment]
What was the motivation for moving away from the secrets?

**Comment by ibotty** [comment]
It did not work for some people, I guess because of bugs in openshift.
Additionally, having the certificates in two places (the route and
a secret) is not nice. I really do hope, that routes when they get
rebased on top of ingresses additionally have a way to reference
a secret.

**Comment by ibotty** [comment]
I don't have much time now, can you please test the path-routes branch? It should fix that part (and lay the foundation of grouping certificates).

**Comment by ibotty** [comment]
Did you have a chance to test the branch?
